File: C++/11.外观模式/FacadePatternDemo.cpp
/**
 * @file FacadePatternDemo.cpp
 * @author {Layne} ({shu_huanglei@163.com})
 * @brief
 * @version 0.1
 * @date 2019-11-09
 *
 * @copyright Copyright (c) 2019
 *
 */
#include "ShapeMaker.h"

int main(int argc, char const *argv[]) {
    ShapeMaker shape_maker;
    shape_maker.DrawCircle();
    shape_maker.DrawRectangle();
    shape_maker.DrawSquare();
    return 0;
}

File: Objective-C/11.FacadePattern/FacadePatternDemo.m
//
//  FacadePatternDemo.m
//  11.FacadePattern
//
//  Created by Layne on 2020/10/29.
//

#import "ShapeMaker.h"

int main(int argc, const char* argv[]) {
    @autoreleasepool {
        ShapeMaker* shapeMaker = [[ShapeMaker alloc] init];
        [shapeMaker drawCircle];
        [shapeMaker drawRectangle];
        [shapeMaker drawSquare];
    }
    return 0;
}
